OpenTelemetry.DotNet.Auto.psm1

```diff
@@ -235,6 +243,14 @@ function Uninstall-OpenTelemetryCore() {
         throw "OpenTelemetry Core is already removed."
     }
 
+    # Unregister .NET Framwework dlls from GAC
+    [System.Reflection.Assembly]::Load("System.EnterpriseServices, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a")            
+    $publish = New-Object System.EnterpriseServices.Internal.Publish 
+    $dlls = Get-ChildItem -Path $installDir\netfx\ -Filter *.dll -File
+    foreach ($dll in $dlls) {
+        $publish.GacRemove($dll)
```

Comment:
This can be a problem with the GAC because by itself it doesn't keep some kind of reference counting, so we may end up removing something added by other applications. I think MSIs interact with some low-level Windows APIs to do some kind of reference counting.

Short of using an MSI we may want to look into whatever API MSIs use for this purpose. That said we should treat as a separate issue and address it in a separate issue/PR.

Reply:
There is a chance that `$publish.GacRemove($dll)` has some logic as `netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51, processorArchitecture=MSIL` has not been removed by this line 😉

Reply:
🤞

Reply:
For the record: assemblies added to the GAC by MSIs aren't removed by `$publish.GacRemove`. Assemblies added to the GAC via `$publish.GacInstall` are not protected in the same way and multiple additions from different locations for the same assembly are clobbered by a single `$publish.GacRemove`.

Reply:
Some more info related to the GAC usage at https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/1906#issuecomment-1376292814